File: onebusaway-android/src/google/java/org/onebusaway/android/map/googlemapsv2/MarkerListeners.java
/*
 * Copyright (C) 2017 Rodrigo Carvalho (carvalhorr@gmail.com)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.onebusaway.android.map.googlemapsv2;

import com.google.android.gms.maps.model.LatLng;
import com.google.android.gms.maps.model.Marker;

/**
 *
 * Interface that each Overlayy needs to impement in order to be notified when a marker is clicked
 * or the map is clicked.
 */

public interface MarkerListeners {
    boolean  markerClicked(Marker marker);
    void removeMarkerClicked(LatLng latLng);
}

File: onebusaway-android/src/google/java/org/onebusaway/android/map/googlemapsv2/MarkerListeners.kt
/* Copyright (C) 2017 Rodrigo Carvalho (carvalhorr@gmail.com) */
package org.onebusaway.android.map.googlemapsv2

import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.Marker

/** Marker and map click callbacks implemented by map overlays. */
interface MarkerListeners {
    fun markerClicked(marker: Marker): Boolean

    fun removeMarkerClicked(latLng: LatLng)
}
